src/components/messages/MsgSigners/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState, type ReactElement } from 'react'
+import { Box, Link, List, ListItem, ListItemIcon, ListItemText } from '@mui/material'
+import FiberManualRecordIcon from '@mui/icons-material/FiberManualRecord'
+import AddCircleIcon from '@mui/icons-material/AddCircle'
+import CheckCircleIcon from '@mui/icons-material/CheckCircle'
+
+import useWallet from '@/hooks/wallets/useWallet'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import useIsMsgPending from '@/hooks/useIsMsgPending'
+import useIsMessageSignableBy from '@/hooks/useIsMsgSignableBy'
+import type { Message } from '@/hooks/useMessages'
+
+import css from '@/components/messages/MsgSigners/styles.module.css'
+
+// TODO: Migrate TxSigners to no longer use a Stepper and move CSS there
+import txSignersCss from '@/components/transactions/TxSigners/styles.module.css'
+
+// Icons
+
+const DotIcon = () => <FiberManualRecordIcon className={css.dot} />
+
+const shouldHideConfirmations = (msg: Message): boolean => {
+  const confirmationsNeeded = msg.confirmationsRequired - msg.confirmationsSubmitted
+  const isConfirmed = confirmationsNeeded <= 0
+
+  // Threshold reached or more than 3 confirmations
+  return isConfirmed || msg.confirmations.length > 3
+}
+
+export const MsgSigners = ({ msg }: { msg: Message }): ReactElement => {
+  const [hideConfirmations, setHideConfirmations] = useState<boolean>(shouldHideConfirmations(msg))
+  const isPending = useIsMsgPending(msg.messageHash)
+  const wallet = useWallet()
+
+  const toggleHide = () => {
+    setHideConfirmations((prev) => !prev)
+  }
+
+  const { confirmations, confirmationsRequired, confirmationsSubmitted } = msg
+
+  const canSign = useIsMessageSignableBy(msg, wallet?.address || '')
+  const confirmationsNeeded = confirmationsRequired - confirmationsSubmitted
+  const isConfirmed = confirmationsNeeded <= 0 || isPending || canSign
```

Comment:
Checking for `canSign` here is a bit confusing. If the wallet can (has to) still sign the message why would `isConfirmed` be true? Shouldn't it only be confirmed if all owners have signed?

Reply:
Valid point. I've adjusted the logic to only rely on whether it's confirmed (using the same logic as your previous comment).